### PR TITLE
sip2: implements selfcheck renewal

### PR DESCRIPTION
**Pull request by lauren-d**
* Allows the patron to do a renewal on the selfcheck machine.
* Adds CLI for selfcheck terminal creation.
* Adds renew handler..

Lauren-D <laurent.dubois@itld-solutions.be>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by rerowep — APPROVED** [review]

File: rero_ils/modules/selfcheck/api.py [hunk 1]
```diff
@@ -427,3 +429,70 @@ def selfcheck_checkin(transaction_user_pid, item_barcode, **kwargs):
                 _('Error encountered: please contact a librarian'))
             raise SelfcheckCirculationError('self checkin failed', checkin)
         return checkin
+
+
+def selfcheck_renew(transaction_user_pid, item_barcode, **kwargs):
+    """SIP2 handler to perform renew.
+
+    perform renew action received from the selfcheck.
```
Comment (iGormilhit) [inline rero_ils/modules/selfcheck/api.py:437]:
```suggestion
    Perform renew action received from the selfcheck.
```
I'm not sure about it, but I would say it's a sentence.

File: rero_ils/modules/selfcheck/cli.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2019 RERO
+# Copyright (C) 2020 UCLouvain
```
Comment (iGormilhit) [inline rero_ils/modules/selfcheck/cli.py:5]:
```suggestion
# Copyright (C) 2021 RERO
# Copyright (C) 2021 UCLouvain
```

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message proposition:

```
sip2: implements selfcheck renewal

* Allows the patron to do a renewal on the selfcheck machine.
* Adds CLI for selfcheck terminal creation.
* Adds renew handler.
```

Is "terminal" explicit enough? Is it to create a session, a user, I don't know whtat… But a terminal sound like an actual device to me.

File: rero_ils/modules/selfcheck/api.py [hunk 3]
```diff
@@ -427,3 +429,70 @@ def selfcheck_checkin(transaction_user_pid, item_barcode, **kwargs):
                 _('Error encountered: please contact a librarian'))
             raise SelfcheckCirculationError('self checkin failed', checkin)
         return checkin
+
+
+def selfcheck_renew(transaction_user_pid, item_barcode, **kwargs):
+    """SIP2 handler to perform renew.
```
Comment (jma) [inline rero_ils/modules/selfcheck/api.py:435]: params descriptions are missing

File: rero_ils/modules/selfcheck/api.py [hunk 4]
```diff
@@ -427,3 +429,70 @@ def selfcheck_checkin(transaction_user_pid, item_barcode, **kwargs):
...
+                    # do extend loan
+                    result, data = item.extend_loan(
+                        transaction_user_pid=staffer.pid,
+                        transaction_library_pid=terminal.library_pid,
+                        item_pid=item.pid,
+                        selfcheck_terminal_id=str(terminal.id),
+                    )
+                    if data[LoanAction.EXTEND]:
+                        loan_pid = data[LoanAction.EXTEND].get('pid')
+                        loan = Loan.get_record_by_pid(loan_pid)
```
Comment (jma) [inline rero_ils/modules/selfcheck/api.py:474]: Sure that you need to get the loan again? Is it not returned by the `data[LoanAction.EXTEND]`?

**Review by jma — APPROVED** [review]

File: rero_ils/modules/selfcheck/api.py (hunk 1 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py]: done

File: rero_ils/modules/selfcheck/cli.py (hunk 2 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/cli.py]: oops...

File: rero_ils/modules/selfcheck/api.py (hunk 3 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py:434]: done

File: rero_ils/modules/selfcheck/api.py (hunk 4 above)
Comment (lauren-d) [reply rero_ils/modules/selfcheck/api.py]: you are right. thanks

**Review by iGormilhit — APPROVED** [review]

**Pull request merged**